### PR TITLE
[Command] [SQL] Add show database/table/partition id command

### PR DESCRIPTION
**Pull request by xinghuayu007**
## Proposed changes

In BE, when a problem happened, in the log, we can find the database id, table id, partition id, but no database name, table name, partition name.

_2021-05-18 19:07:52,331 INFO 193033 [DatabaseTransactionMgr.commitTransaction():512] transaction:[TransactionState. transaction id: 285498579, label: index_subtask_0_time_1621336072089_count_1_48cfe999-7460-439e-a284-d71b2943e203, db id: 19595062, table id list: 169245759, callback id: -1, coordinator: BE: host:51692, transaction status: COMMITTED, error replicas num: 0, replica ids: , prepare time: 1621336072307, commit time: 1621336072329, finish time: -1, reason: ] successfully committed_

 In FE, there also no way to find database name/table name/partition name accourding to database id/table id/partition id. Therefore, this patch add 3 new commands:
1. show database id;
mysql> show database 10002;
+----------------------+
| DbName               |
+----------------------+
| default_cluster:test |
+----------------------+
2. show table id;
mysql> show table 11100;
+----------------------+-----------+-------+
| DbName               | TableName | DbId  |
+----------------------+-----------+-------+
| default_cluster:test | table2    | 10002 |
+----------------------+-----------+-------+

3. show partition id;
mysql> show partition 11099;
+----------------------+-----------+---------------+-------+---------+
| DbName               | TableName | PartitionName | DbId  | TableId |
+----------------------+-----------+---------------+-------+---------+
| default_cluster:test | table2    | p201708       | 10002 | 11100   |
+----------------------+-----------+---------------+-------+---------+
## Types of changes

What types of changes does your code introduce to Doris?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)
- [ ] Code refactor (Modify the code structure, format the code, etc...)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [ ] I have created an issue on (Fix #ISSUE) and described the bug/feature there in detail
- [x] Compiling and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] If these changes need document changes, I have updated the document
- [x] Any dependent changes have been merged

## Further comments

If this is a relatively large or complex change, kick off the discussion at dev@doris.apache.org by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: fe/fe-core/src/main/java/org/apache/doris/analysis/ShowDbIdStmt.java [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+import org.apache.doris.catalog.Column;
+import org.apache.doris.catalog.ScalarType;
+import org.apache.doris.common.AnalysisException;
+import org.apache.doris.common.ErrorCode;
+import org.apache.doris.common.ErrorReport;
+import org.apache.doris.mysql.privilege.PrivPredicate;
+import org.apache.doris.qe.ConnectContext;
+import org.apache.doris.qe.ShowResultSetMetaData;
+
+// SHOW TABLES
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/analysis/ShowDbIdStmt.java:30]: comment is wrong

File: fe/fe-core/src/main/java/org/apache/doris/analysis/ShowPartitionIdStmt.java [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+import org.apache.doris.catalog.Column;
+import org.apache.doris.catalog.ScalarType;
+import org.apache.doris.common.AnalysisException;
+import org.apache.doris.common.ErrorCode;
+import org.apache.doris.common.ErrorReport;
+import org.apache.doris.mysql.privilege.PrivPredicate;
+import org.apache.doris.qe.ConnectContext;
+import org.apache.doris.qe.ShowResultSetMetaData;
+
+// SHOW TABLES
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/analysis/ShowPartitionIdStmt.java:30]: comment

File: fe/fe-core/src/main/java/org/apache/doris/analysis/ShowTableIdStmt.java [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+    }
+
+    @Override
+    public void analyze(Analyzer analyzer) throws AnalysisException {
+        // check access first
+        if (!Catalog.getCurrentCatalog().getAuth().checkGlobalPriv(ConnectContext.get(), PrivPredicate.ADMIN)) {
+            ErrorReport.reportAnalysisException(ErrorCode.ERR_SPECIFIC_ACCESS_DENIED_ERROR, "SHOW TABLE");
+        }
+        if (Strings.isNullOrEmpty(db)) {
+            db = analyzer.getDefaultDb();
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/analysis/ShowTableIdStmt.java:57]: Do we need to check db here?
I think the purpose of this stmt is to find out which database is this table belongs to.

File: fe/fe-core/src/main/java/org/apache/doris/qe/ShowExecutor.java [hunk 4]
```diff
@@ -449,6 +458,67 @@ private void handleShowMigrations() throws AnalysisException {
...
+        resultSet = new ShowResultSet(showStmt.getMetaData(), rows);
+    }
+
+    private void handleShowTableId() throws AnalysisException {
+        ShowTableIdStmt showStmt = (ShowTableIdStmt) stmt;
+        long tableId = showStmt.getTableId();
+        List<List<String>> rows = Lists.newArrayList();
+        Catalog catalog = ctx.getCatalog();
+        List<Long> dbIds = catalog.getDbIds();
+        for (long dbId:dbIds) {
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/qe/ShowExecutor.java:481]:
```suggestion
        for (long dbId : dbIds) {
```
format

File: fe/fe-core/src/main/java/org/apache/doris/qe/ShowExecutor.java [hunk 5]
```diff
@@ -449,6 +458,67 @@ private void handleShowMigrations() throws AnalysisException {
...
+            Database database = catalog.getDb(dbId);
+            Table table = database.getTable(tableId);
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/qe/ShowExecutor.java:483]: database maybe null here, cause this is not an atomic operation

File: fe/fe-core/src/main/java/org/apache/doris/qe/ShowExecutor.java [hunk 6]
```diff
@@ -449,6 +458,67 @@ private void handleShowMigrations() throws AnalysisException {
...
+            if (table != null) {
+                List<String> row = new ArrayList<>();
+                row.add(database.getFullName());
+                row.add(table.getName());
+                row.add(String.valueOf(database.getId()));
+                rows.add(row);
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/qe/ShowExecutor.java:489]: I think we should break here, cause each table has a unique id.

File: fe/fe-core/src/main/java/org/apache/doris/qe/ShowExecutor.java [hunk 7]
```diff
@@ -449,6 +458,67 @@ private void handleShowMigrations() throws AnalysisException {
...
+    private void handleShowPartitionId() throws AnalysisException {
+        ShowPartitionIdStmt showStmt = (ShowPartitionIdStmt) stmt;
+        long partitionId = showStmt.getPartitionId();
+        List<List<String>> rows = Lists.newArrayList();
+        Catalog catalog = ctx.getCatalog();
+        List<Long> dbIds = catalog.getDbIds();
+        for (long dbId:dbIds) {
+            Database database = catalog.getDb(dbId);
+            List<Table> tables = database.getTables();
+            for (Table tbl:tables) {
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/qe/ShowExecutor.java:504]:
```suggestion
            for (Table tbl : tables) {
```
format

And we need tbl's readlock here.

File: fe/fe-core/src/main/java/org/apache/doris/qe/ShowExecutor.java [hunk 8]
```diff
@@ -449,6 +458,67 @@ private void handleShowMigrations() throws AnalysisException {
...
+                if (tbl instanceof OlapTable) {
+                    Partition partition = ((OlapTable) tbl).getPartition(partitionId);
+                    if (partition != null) {
+                        List<String> row = new ArrayList<>();
+                        row.add(database.getFullName());
+                        row.add(tbl.getName());
+                        row.add(partition.getName());
+                        row.add(String.valueOf(database.getId()));
+                        row.add(String.valueOf(tbl.getId()));
+                        rows.add(row);
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/qe/ShowExecutor.java:514]: break out the very outside loop.

**Review by morningman — COMMENTED** [review]
Please add document for this

**Comment by xinghuayu007** [comment]
> Please add document for this

ｄｏｎｅ

File: fe/fe-core/src/main/java/org/apache/doris/qe/ShowExecutor.java [hunk 9]
```diff
@@ -449,6 +458,81 @@ private void handleShowMigrations() throws AnalysisException {
...
+            Table table = database.getTable(tableId);
+            if (table != null) {
+                List<String> row = new ArrayList<>();
+                row.add(database.getFullName());
+                row.add(table.getName());
+                row.add(String.valueOf(database.getId()));
+                rows.add(row);
+                break;
+            }
+            database.readUnlock();
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/qe/ShowExecutor.java:496]: There is no need to lock the database.
And btw, if it is necessary, you should use `try...finally` to do this.

File: fe/fe-core/src/main/java/org/apache/doris/qe/ShowExecutor.java [hunk 10]
```diff
@@ -449,6 +458,81 @@ private void handleShowMigrations() throws AnalysisException {
...
+        long partitionId = showStmt.getPartitionId();
+        List<List<String>> rows = Lists.newArrayList();
+        Catalog catalog = ctx.getCatalog();
+        List<Long> dbIds = catalog.getDbIds();
+        for (long dbId : dbIds) {
+            Database database = catalog.getDb(dbId);
+            if (database == null) {
+                continue;
+            }
+            database.readLock();
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/qe/ShowExecutor.java:512]: No need database lock

File: fe/fe-core/src/main/java/org/apache/doris/qe/ShowExecutor.java (hunk 9 above)
Comment (xinghuayu007) [reply fe/fe-core/src/main/java/org/apache/doris/qe/ShowExecutor.java]: updated

**Review by morningman — APPROVED** [review]
LGTM

**Pull request merged**